### PR TITLE
EZP-1308: Update design for custom tags popups

### PR DESCRIPTION
**Pull request by GrabowskiM**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | https://jira.ez.no/browse/EZP-31308
| **Improvement**| yes
| **New feature**    | no
| **Target version** | master
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

**Comment by GrabowskiM** [comment]
@alongosz I've added here bazinga script and yaml file, but configuration `alloy_editor` is already defined in admin_ui package, @dew326 asked me to ping you if we can remove it from admin_ui and leave it only here

File: src/bundle/Resources/views/themes/standard/ezrichtext/custom_tags/ezyoutube.html.twig [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <div class="embed-container" {% if align is defined %} style="text-align: {{ align }};"{% endif %}>
+{{dump(params)}}
```
Comment (alongosz) [inline src/bundle/Resources/views/themes/standard/ezrichtext/custom_tags/ezyoutube.html.twig:2]: debug code detected :P

File: src/bundle/Resources/public/scss/functions/checkbox-switcher.scss [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        &:after {
+            right: 0;
+            transform: translate(50%, -50%);
+        }
+
+        &.is-disabled {
+            opacity: 0.5;
+        }
+    }
+}
```
Comment (alongosz) [inline src/bundle/Resources/public/scss/functions/checkbox-switcher.scss:72]: missing new line

File: src/bundle/DependencyInjection/EzPlatformRichTextExtension.php [hunk 3]
```diff
@@ -138,6 +140,34 @@ private function prependEzRichTextConfiguration(ContainerBuilder $container): vo
...
+        $container->prependExtensionConfig('jms_translation', [
+            'configs' => [
+                'ezplatform_richtext' => [
+                    'dirs' => [
+                        __DIR__ . '/../../../src/',
+                    ],
+                    'output_dir' => __DIR__ . '/../Resources/translations/',
+                    'output_format' => 'xliff',
+                    'excluded_dirs' => ['Behat', 'Tests', 'node_modules'],
+                    'extractors' => ['ez_policy', 'ez_policy_limitation', 'ez_location_sorting'],
```
Comment (alongosz) [inline src/bundle/DependencyInjection/EzPlatformRichTextExtension.php:165]: why would RichText define  those extractors?

**Comment by dew326** [comment]
Please add some screenshots

File: src/bundle/Resources/public/js/OnlineEditor/buttons/ez-btn-customtag-update.js [hunk 4]
```diff
@@ -156,7 +163,11 @@ export default class EzBtnCustomTagUpdate extends EzWidgetButton {
         const renderMethods = this.getAttributeRenderMethods();
         const methodName = renderMethods[attributeConfig.type];
 
-        return <div className="ez-ae-custom-tag__attributes">{this[methodName](attributeConfig, attribute)}</div>;
+        return (
+            <div className={`ez-ae-custom-tag__attributes ez-ae-custom-tag__attribute-type--${attributeConfig.type} ez-ae-custom-tag__attribute-name--${attribute}`}>
```
Comment (dew326) [inline src/bundle/Resources/public/js/OnlineEditor/buttons/ez-btn-customtag-update.js:167]: This doesn't seem like proper BEM.

File: src/bundle/Resources/public/scss/_alloyeditor-ez.scss [hunk 5]
```diff
@@ -1092,9 +1140,89 @@ pre.ez-has-anchor > .ez-icon {
...
+                margin-right: calculateRem(5px);
+            }
+
+            .btn-primary {
+                color: $ez-ae-color-primary;
+                margin-left: calculateRem(5px);
+            }
+
+            &--ezyoutube {
+                width: calculateRem(280px);
```
Comment (dew326) [inline src/bundle/Resources/public/scss/_alloyeditor-ez.scss:1155]: Why do you specify width for our custom tags? I think it should be general rule.

File: src/bundle/Resources/translations/ezplatform_content_forms_policies.en.xliff [hunk 6]
```diff
@@ -0,0 +1,101 @@
+<?xml version="1.0" encoding="utf-8"?>
+<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:jms="urn:jms:translation" version="1.2">
+  <file source-language="en" target-language="en" datatype="plaintext" original="not.available">
+    <header>
+      <tool tool-id="JMSTranslationBundle" tool-name="JMSTranslationBundle" tool-version="1.1.0-DEV"/>
+      <note>The source node in most cases contains the sample message as written by the developer. If it looks like a dot-delimitted string such as "form.label.firstname", then the developer has not provided a default message.</note>
+    </header>
+    <body>
+      <trans-unit id="3742f2cc514848bd4a5eb03430e412d44f5d9ec2" resname="policy.limitation.identifier.class">
+        <source>Class</source>
```
Comment (DominikaK) [inline src/bundle/Resources/translations/ezplatform_content_forms_policies.en.xliff:10]: I believe all the labels from this file are used for the same cases as in https://github.com/ezsystems/ezplatform-admin-ui/blob/master/src/bundle/Resources/translations/ezplatform_content_forms_policies.en.xliff, so they should be the same

File: src/bundle/Resources/translations/content_type.en.xliff [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  <file source-language="en" target-language="en" datatype="plaintext" original="not.available">
+    <header>
+      <tool tool-id="JMSTranslationBundle" tool-name="JMSTranslationBundle" tool-version="1.1.0-DEV"/>
+      <note>The source node in most cases contains the sample message as written by the developer. If it looks like a dot-delimitted string such as "form.label.firstname", then the developer has not provided a default message.</note>
+    </header>
+    <body>
+      <trans-unit id="c625b80940dd9f13842e38446a97391248d0f709" resname="content_type.sort_field.1">
+        <source>Location path</source>
+        <target state="new">Location path</target>
+        <note>key: content_type.sort_field.1</note>
```
Comment (dew326) [inline src/bundle/Resources/translations/content_type.en.xliff:12]: Doesn't seem like this should be part of richtext

File: src/bundle/Resources/translations/ezplatform_content_forms_policies.en.xliff [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        <target>Class</target>
+        <note>key: policy.limitation.identifier.class</note>
```
Comment (dew326) [inline src/bundle/Resources/translations/ezplatform_content_forms_policies.en.xliff:12]: Also not sure if it should be here

File: src/bundle/Resources/translations/forms.en.xliff [hunk 9]
```diff
@@ -0,0 +1,381 @@
...
+  <file source-language="en" target-language="en" datatype="plaintext" original="not.available">
+    <header>
+      <tool tool-id="JMSTranslationBundle" tool-name="JMSTranslationBundle" tool-version="1.1.0-DEV"/>
+      <note>The source node in most cases contains the sample message as written by the developer. If it looks like a dot-delimitted string such as "form.label.firstname", then the developer has not provided a default message.</note>
+    </header>
+    <body>
+      <trans-unit id="d29c07272fff793138cbb6f7dbb41b08a16ff815" resname="role.policy.all_modules">
+        <source>All modules</source>
+        <target>All modules</target>
+        <note>key: role.policy.all_modules</note>
```
Comment (dew326) [inline src/bundle/Resources/translations/forms.en.xliff:12]: The same

File: src/bundle/Resources/translations/ezplatform_content_forms_policies.en.xliff (hunk 6 above)
Comment (alongosz) [reply src/bundle/Resources/translations/ezplatform_content_forms_policies.en.xliff:10]: > I believe all the labels from this file are used for the same cases as in https://github.com/ezsystems/ezplatform-admin-ui/blob/master/src/bundle/Resources/translations/ezplatform_content_forms_policies.en.xliff, so they should be the same

I actually don't think `ezplatform_content_forms_policies.en.xliff` belongs in this package.

**Comment by GrabowskiM** [comment]
Screenshot from 2020-05-12 11-33-30
Screenshot from 2020-05-12 11-34-58
Screenshot from 2020-05-12 11-35-28

File: src/bundle/Resources/public/js/OnlineEditor/buttons/ez-btn-customtag-update.js [hunk 10]
```diff
@@ -56,18 +56,25 @@ export default class EzBtnCustomTagUpdate extends EzWidgetButton {
...
-                <input
-                    type="checkbox"
-                    defaultValue={config.defaultValue}
-                    required={config.required}
-                    className="attribute__input form-control"
-                    checked={this.state.values[attrName].value}
-                    onChange={this.updateValues.bind(this)}
-                    data-attr-name={attrName}
-                />
+                <div class="ez-data-source" title="">
```
Comment (dew326) [inline src/bundle/Resources/public/js/OnlineEditor/buttons/ez-btn-customtag-update.js:64]: This bundle has no idea about admin-ui, you cannot use classes from admin-ui

**Comment by GrabowskiM** [comment]
> Makes sense to keep it in RichText only, but it seems like out of scope. Is it a step needed for this task to be completed?

Unfortunately yes, we need to add here these scripts, as I've added new translation and had to extract it

**Comment by alongosz** [comment]
> > Makes sense to keep it in RichText only, but it seems like out of scope. Is it a step needed for this task to be completed?
> 
> Unfortunately yes, we need to add here these scripts, as I've added new translation and had to extract it

Ok, then we need to prepare history. Given it's been already reviewed & approved, you can rebase it and split the changes into two separate commits:
- `Prepended JMS and Bazinga configuration`
- `EZP-1308: Updated design for custom tags popups`

(the change to `src/bundle/Resources/translations/alloy_editor.en.xliff` probably belongs to the scope of the latter one).

I've marked the PR for fast-forward merge. Let me know once it's ready for merge, I can handle it.
#cleanhistory

**Comment by dew326** [comment]
For QA: In `ezyoutube` tag, there are two new attributes `width` and `height`, please test it.

**Comment by bogusez** [comment]
@GrabowskiM 
Selected value isn't visible for user for custom tag 'twitter' in dropdown 'cards' and 'conversation'.
Screenshot 2020-05-15 at 10 33 27
Screenshot 2020-05-15 at 10 33 16

**Comment by bogusez** [comment]
Although custom tag is used in field 'intro' the validation still says that 'intro field is required'. 
Btw I know it's not a scope of this task but I found it while testing design improvements.
Screenshot 2020-05-15 at 10 34 42

**Comment by bogusez** [comment]
Edit button won't display edit options of selected custom tag (user can't edit custom tag settings).
Screenshot 2020-05-15 at 12 41 09

**Pull request merged**